ONSdigital/ras-party
# Remove bi/send no active enrolment event

# Motivation and Context
When no active enrolments are remaining for a business/survey we need to updated cases in the case service. Currently we send a `DISABLE_RESPONDENT_ENROLMENT` case event against each `BI` case we are disabling. As we are moving toward the removal of `BI` cases we need an equivalent which will work for `B` cases.

# What has changed
A new `NO_ACTIVE_ENROLMENTS` case event is sent to case service when the final respondent is disabled for a business/survey. 
We continue sending `DISABLE_RESPONDENT_ENROLMENT` case events against the associated `BI` cases as these have not yet been removed from the system

# How to test?
Disable the final respondent for a business/survey either through the UI or the party service api, confirm that a case event with category `NO_ACTIVE_ENROLMENTS` is posted to the case service
See associated [case-service PR]() for testing instructions with all associated PR's

Unit tests should pass
Acceptance tests should pass when run alongside associated PR's

# Links
[Trello](https://trello.com/c/RlGDFpX8/18-disable-enrolment-stop-creating-b-cases-case-events-l)
### Associated PR's
[case-service](https://github.com/ONSdigital/rm-case-service/pull/59)
[casesvc-api](https://github.com/ONSdigital/rm-casesvc-api/pull/25)

!!! This shouldn't be deployed into production until the appropriate data migration has been done to ensure that one B case exists for every casegroup !!!
